asset/src/spaces_reader_api/schema.ts

```diff
@@ -1,15 +1,59 @@
-import { ConvictSchema, AnyObject, cloneDeep } from '@terascope/job-components';
-import { spacesSchema } from '../spaces_reader/schema';
+import {
+    ConvictSchema,
+    AnyObject,
+    getTypeOf,
+    cloneDeep,
+    isNumber,
+    ValidatedJobConfig
+} from '@terascope/job-components';
+import elasticAPI from '@terascope/elasticsearch-api';
+import { schema } from '../elasticsearch_reader_api/schema';
 import { ApiConfig } from '../elasticsearch_reader/interfaces';
+import { DEFAULT_API_NAME } from './interfaces';
+
+const clone = cloneDeep(schema);
+
+const apiSchema = {
+    endpoint: {
+        doc: 'The base API endpoint to read from: i.e. http://yourdomain.com/api/v1',
+        default: null,
+        format: 'required_String'
+    },
+    token: {
+        doc: 'API access token for making requests',
+        default: null,
+        format: 'required_String'
+    },
+    timeout: {
+        doc: 'Time in milliseconds to wait for a connection to timeout.',
+        default: 300000,
+        format(val: unknown): void {
+            if (!isNumber(val)) throw new Error(`Invalid parameter timeout, it must be of type number, was given ${getTypeOf(val)}`);
+            if (isNaN(val)) throw new Error('Invalid timeout parameter, must be a number');
+            if (val <= 0) throw new Error('Invalid timeout parameter, must be greater than zero');
+        }
+    }
+};
+
+const spacesSchema = Object.assign({}, clone, apiSchema) as AnyObject;
 
-const clone = cloneDeep(spacesSchema) as AnyObject;
 // this should not continue onward
-delete clone.api_name;
+delete spacesSchema.api_name;
 
-clone.index.format = 'required_String';
+spacesSchema.date_field_name.format = 'required_String';
```

Comment:
I don't think this is required?

Reply:
It is, this is a facade over the elasticsearch_reader_api (which schema we use to build this) however we do not support the id_reader side of it. Thus it only will work for the date side of queries, so it is required

Reply:
Oh I guess that makes sense